runtime/src/bank.rs

```diff
@@ -3716,6 +3716,7 @@ impl Bank {
                                 loaded_accounts_count: loaded_accounts.len(),
                                 loaded_accounts_data_size,
                             },
+                            fee_payer_post_balance: loaded_accounts[0].1.lamports(),
```

Comment:
This balance might not be correct for failed transactions. Similar to fees-only tx's you have to use the rollback accounts fee payer balance for tx's that fail execution.